lib/facility.rb

```diff
@@ -1,14 +1,58 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :collected_fees,
+              :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle) 
+    if @services.include? 'Vehicle Registration'
+      @registered_vehicles << vehicle
+      collect_fee(vehicle)
+      vehicle.register
+      vehicle.get_plate
```

Comment:
Nice job with all of these helper methods! You could also have encapsulated line 24 into a helper method to have the "job" of register_vehicle just be to delegate tasks to other helper methods.